### DISCUSSION TITLE
CV-CUDA可以在 arm架构下运行么？例如 Orin或者更早的NX？

### DISCUSSION THREAD

**Discussion by cmffire** [CVCUDA/CV-CUDA#32] [Q&A]
如题。有相关例子么？

**Comment by kgnandanwar** [comment]
# CV-CUDA on ARM Architecture (Jetson Orin/NX)

Yes, CV-CUDA runs on ARM, including Jetson devices.

## Supported ARM Platforms

**Jetson Orin (AGX Orin, IGX Orin):**
- ✅ Fully supported
- Use `aarch64_cu12` build
- Requires JetPack 6.0 DP with driver r535
- Python 3.10

**Older Jetson (Xavier NX, Nano, etc.):**
- ❌ Not officially supported
- Requires compute capability ≥ SM7.5 (Orin is SM8.7)
- Xavier NX is SM7.2 - below minimum requirement

## Installation on Jetson Orin

**Option 1: Python wheel (easiest)**
```bash
# On Jetson Orin with JetPack 6.0
pip install cvcuda-cu12
```

**Option 2: Build from source**
```bash
git clone https://github.com/CVCUDA/CV-CUDA.git
cd CV-CUDA
./ci/build.sh
```

## Platform Support Summary (from README.md)

| Platform | CUDA | Hardware | Driver | Status |
|----------|------|----------|--------|--------|
| Jetson AGX Orin | 12.2 | SM8.7 (Ampere) | JetPack 6.0, r535 | ✅ Supported |
| Jetson Xavier NX | - | SM7.2 (Volta) | - | ❌ Below SM7.5 requirement |
| Jetson Nano | - | SM5.3 (Maxwell) | - | ❌ Below SM7.5 requirement |

## Quick Test on Jetson

```python
import cvcuda
import torch

# Create a test image
img = cvcuda.Image((640, 480), cvcuda.Format.RGB8)

# Or wrap existing GPU tensor
gpu_tensor = torch.zeros((480, 640, 3), device="cuda", dtype=torch.uint8)
img = cvcuda.as_image(gpu_tensor, format=cvcuda.Format.RGB8)

print(f"Image created: {img.size}")
```

## References

- [README.md](README.md) - Full platform compatibility matrix (line 75)
- Pre-built wheels: [cvcuda-cu12 on PyPI](https://pypi.org/project/cvcuda-cu12/)
- GitHub releases have explicit "Jetson" archives for aarch64 builds

## Bottom Line

**Orin:** Yes, works out of the box.
**NX/Nano:** No, compute capability too old (need SM7.5+, they have SM7.2/5.3).
